File: frontend/src/services/completionService.js
import api from './api';

const completionService = {
    updateProgress: (enrollmentId, data) => api.patch(`/learning/completions/${enrollmentId}/progress`, data),
    getByEnrollmentId: (enrollmentId) => api.get(`/learning/completions/${enrollmentId}`),
    getByEmployeeId: (employeeId) => api.get(`/learning/completions/employee/${employeeId}`),
    getByCourseId: (courseId) => api.get(`/learning/completions/course/${courseId}`),
    getAllCompleted: () => api.get('/learning/completions/completed'),
};

export default completionService;

File: skillsphere-learning-service/src/main/java/com/infosis/nexus/learning/service/CompletionService.java
package com.infosis.nexus.learning.service;

import com.infosis.nexus.learning.dto.CompletionResponse;
import com.infosis.nexus.learning.dto.ProgressUpdateRequest;

import java.util.List;

public interface CompletionService {
    CompletionResponse updateProgress(Long enrollmentId, ProgressUpdateRequest request);
    CompletionResponse getByEnrollmentId(Long enrollmentId);
    List<CompletionResponse> getByEmployeeId(Long employeeId);
    List<CompletionResponse> getByCourseId(Long courseId);
    List<CompletionResponse> getAllCompleted();
}